grails-app/domain/org/pih/warehouse/invoice/InvoiceItem.groovy

```diff
@@ -82,8 +82,17 @@ class InvoiceItem implements Serializable {
         product(nullable: true)
         glAccount(nullable: true)
         budgetCode(nullable: true)
-
-        quantity(nullable: false, min: 0) // min = 0 for canceled items
+        quantity(nullable: false, min: 0, validator: { quantity, obj ->
```

Comment:
could you add types for those arguments? I'm always confused in the validator what is what (I know the names describe it, but still it shouldn't be a pain to change it).